### ISSUE TITLE
Latest v2.3.0 for JDK 23 Pcap.findAllDevs() throws on macOS

### ISSUE THREAD

**Issue by arisona** [slytechs-repos/jnetpcap-wrapper#61]
After upgrading to the latest jnetpcap release v2.3.0, Pcap.findAllDevs() throws on macOS 15.1.1 with JDK 23.0.1, stack trace see below. I remember this was not the case with PR #57, that included some changes. Something must have sneaked into the overall fixes for JDK 23 / release 2.3.0, but I can't easily trace it back.

```
Exception in thread "main" java.lang.IndexOutOfBoundsException: Out of bound access on segment MemorySegment{ address: 0x13d6dc8d0, byteSize: 20 }; new offset = 0; new length = 38
        at java.base/jdk.internal.foreign.AbstractMemorySegmentImpl.outOfBoundException(AbstractMemorySegmentImpl.java:439)
        at java.base/jdk.internal.foreign.AbstractMemorySegmentImpl.apply(AbstractMemorySegmentImpl.java:420)
        at java.base/jdk.internal.foreign.AbstractMemorySegmentImpl.apply(AbstractMemorySegmentImpl.java:70)
        at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:98)
        at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:124)
        at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:448)
        at java.base/jdk.internal.foreign.AbstractMemorySegmentImpl.checkBounds(AbstractMemorySegmentImpl.java:409)
        at java.base/jdk.internal.foreign.AbstractMemorySegmentImpl.checkAccess(AbstractMemorySegmentImpl.java:369)
        at java.base/jdk.internal.foreign.LayoutPath.checkEnclosingLayout(LayoutPath.java:288)
        at org.corebounce.nova/org.jnetpcap.SaLayout.get(SaLayout.java:313)
        at org.corebounce.nova/org.jnetpcap.SaLayout.getNumber(SaLayout.java:325)
        at org.corebounce.nova/org.jnetpcap.SaLayout.getUnsignedShort(SaLayout.java:336)
        at org.corebounce.nova/org.jnetpcap.SockAddr$LinkSockAddr.<init>(SockAddr.java:438)
        at org.corebounce.nova/org.jnetpcap.SockAddr.newInstance(SockAddr.java:713)
        at org.corebounce.nova/org.jnetpcap.PcapIf$PcapAddr.<init>(PcapIf.java:198)
        at org.corebounce.nova/org.jnetpcap.PcapIf$PcapAddr.listAll(PcapIf.java:172)
        at org.corebounce.nova/org.jnetpcap.PcapIf.<init>(PcapIf.java:358)
        at org.corebounce.nova/org.jnetpcap.PcapIf.listAll(PcapIf.java:322)
        at org.corebounce.nova/org.jnetpcap.Pcap0_7.findAllDevs(Pcap0_7.java:204)
        at org.corebounce.nova/org.jnetpcap.Pcap.findAllDevs(Pcap.java:1098)
        ....
```

**Comment by SecureNetWizard** [comment]
Taking a look at it now. I'm able to reproduce the failure it as well.

**Comment by SecureNetWizard** [comment]
Found the problem. Will checkin this afternoon. A very minor issue. I did modify how address structure length is calculated for the JDK 23 implementation. That did not work well with JDK 23 on Mac (being BSD derrived).

**Comment by SecureNetWizard** [comment]
Had to do a little branch cleanup but everything looks great now:

```bash
sdks/jnetpcap-wrapper$ git branch -r
  origin/bugfix/issue-61-v230-for-jdk-23-pcapfindalldevs-throws-on-macos
  origin/develop
  origin/gh-pages
  origin/jdk21/develop
  origin/jdk21/main
  origin/main
```

I originally fixed this while at the airport between flights, but now I'm taking a second closer look at why this failed in the first place, so specifically for BSD/MacOS. 

I will back propagate the changes in the JDK 21 do another point release for JDK 22+ and JDK 21 (v2.3.1 and v2.3.1+jdk21)

**Comment by SecureNetWizard** [comment]
This issue is fixed in `v2.3.1` and `v2.3.1+jdk21`.

Published to maven central (allow upto 24 hrs).
